stdlib/Dates/src/periods.jl

```diff
@@ -371,8 +371,15 @@ Base.convert(::Type{T}, x::CompoundPeriod) where T<:Period =
 (-)(x::CompoundPeriod) = CompoundPeriod(-x.periods)
 (-)(y::Union{Period, CompoundPeriod}, x::CompoundPeriod) = (-x) + y
 
+(*)(x::CompoundPeriod, y::Real) = iszero(y) ? CompoundPeriod() : CompoundPeriod(x.periods * y)
+(*)(y::Real, x::CompoundPeriod) = x * y
+
+(/)(x::CompoundPeriod, y::Real) = isinf(y) ? CompoundPeriod() : CompoundPeriod(x.periods / y)
```

Comment:
I'm not 100% convinced by these `isinf` and `iszero` checks.  This corresponds to canonicalizing the result, no?  But why do we only canonicalize a single case?

Reply:
The `CompoundPeriod` constructor canonicalizes the result anyway, these checks are just improving performance.

Reply:
It's not clear that the performance improvement (for a pretty unusual case) is worth the increase in code complexity here.

Reply:
Okay, I have removed the special handling of the `isinf`/`iszero` cases.